src/XIOPAK.CPP

```diff
@@ -625,18 +631,22 @@ SEC FC xfflush(		// Flush all buffers associated with a file
 
 // Returns SECOK if all OK, else other sec (SETS xf->xflsterr)
 {
-	SI acc;
+	std::string acc;
 	LI fpos;
 
 	if (xf->xflsterr==SECOK)			// if no prior error
 	{
-		acc = xf->access & (O_RDONLY|O_WRONLY|O_RDWR|O_TEXT|O_BINARY);
-		// close and re-open file to flush buffers to disk (not just to OS)
-		if ( (fpos = tell(xf->han)) != -1
-				&& close(xf->han) != -1
-				&& (xf->han = _open( xf->xfname, acc, S_IREAD|S_IWRITE)) != -1
-				&& _lseek( xf->han, fpos, SEEK_SET) != -1 )
-			return SECOK;
+		acc = std::string(xf->access);
```

Comment:
Using a std::string is a bit heavy-handed here (large object, new heap allocation). Since xf->access is simply a char[3], why not use strchr?

Reply:
Cool! I didn't know C-style string had a function like this.

Reply:
I think you used `strcpy` when you changed the access variable from an int to a char[3]. You'll find that they are related. You can look up C functions starting with "str" and discover a treasure trove.